### PR TITLE
15 renaming attachments

### PR DESCRIPTION
**Pull request by ElBabbler**
Add custom filename field to upload file action

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,10 @@ The following configuration fields are available:
 
 * Note: if the directory does not exist, it will create it at the risk of possibly overwriting any files that may have the same name.
 
+Input metadata:
+
+- **Filename**: Custom name for uploadad file.
```
Comment (kirill-levitskiy) [inline README.md]: - **Filename**: Custom name for uploaded file.

File: lib/actions/upload.js [hunk 2]
```diff
@@ -11,22 +11,36 @@ const Sftp = require('../Sftp');
...
   await sftp.connect();
 
   const result = {
     results: [],
   };
   const dir = cfg.directory || '/';
+  // eslint-disable-next-line no-nested-ternary
+  const filename = msg.body.filename
+    ? (Object.keys(msg.attachments).length > 1 ? msg.body.filename.split('.')[0] : msg.body.filename)
+    : null;
```
Comment (uaArsen) [inline lib/actions/upload.js]: extract this code to method with understandable name, right now it looks over complicated

File: lib/actions/upload.js [hunk 3]
```diff
@@ -11,22 +11,36 @@ const Sftp = require('../Sftp');
...
+  this.logger.info(`Prepared filename: ${filename}`);
+
   const isExists = await sftp.exists(dir);
   if (!isExists) {
     await sftp.mkdir(dir, true);
   }
+
+  this.logger.info(`Found ${Object.keys(msg.attachments).length} attachments`);
```
Comment (uaArsen) [inline lib/actions/upload.js]: should it be trace level?

File: lib/actions/upload.js [hunk 4]
```diff
@@ -11,22 +11,36 @@ const Sftp = require('../Sftp');
...
   // eslint-disable-next-line no-restricted-syntax
   for (const key of Object.keys(msg.attachments)) {
     const attachment = msg.attachments[key];
     const cur = await sftp.cwd();
-    const targetPath = (cur.charAt(0) === '/') ? path.posix.resolve(dir, key) : path.resolve(dir, key);
+    // eslint-disable-next-line no-nested-ternary
```
Comment (uaArsen) [inline lib/actions/upload.js]: ??

File: spec-integration/integration.spec.js [hunk 5]
```diff
@@ -65,7 +65,7 @@ describe('SFTP integration test - upload then download', function () {
       },
     };
     const result = await upload.process.call(sender, msg, cfg);
-    
+
```
Comment (uaArsen) [inline spec-integration/integration.spec.js]:
```suggestion
```

File: spec-integration/integration.spec.js [hunk 6]
```diff
@@ -65,7 +65,7 @@ describe('SFTP integration test - upload then download', function () {
...
     expect(result.body.results).to.be.an('array');
     expect(result.body.results.length).to.equal(1);
     expect(result.body.results[0].attachment).to.equal('logo.svg');
```
Comment (uaArsen) [inline spec-integration/integration.spec.js]: Do we have tests for this feature?

File: lib/actions/upload.js (hunk 3 above)
Comment (ElBabbler) [reply lib/actions/upload.js]: no - it's just number to check basic logic processing

File: spec-integration/integration.spec.js (hunk 6 above)
Comment (ElBabbler) [reply spec-integration/integration.spec.js]: Nope - it's default eslint fix

File: lib/actions/upload.js (hunk 2 above)
Comment (ElBabbler) [reply lib/actions/upload.js]: Other way is complicated if-else structure - ternary operator is simpier to read

File: spec/triggers/read.spec.js [hunk 7]
```diff
@@ -12,9 +12,9 @@ require('dotenv').config();
 
 describe('SFTP', () => {
   const sftp = new Sftp(bunyan.createLogger({ name: 'dummy' }), {
-    host: process.env.HOSTNAME,
+    host: process.env.SFTP_HOSTNAME,
     port: process.env.PORT,
```
Comment (kirill-levitskiy) [inline spec/triggers/read.spec.js]: Please, replace it with `port: Number(process.env.PORT),` as it has been implemented in `spec-integration/verifyCredentials.spec.js`

Comment (ElBabbler) [inline spec/triggers/read.spec.js]:
```suggestion
    port: Number(process.env.PORT),
```

File: spec/triggers/read.spec.js [hunk 8]
```diff
@@ -12,9 +12,9 @@ require('dotenv').config();
 
 describe('SFTP', () => {
   const sftp = new Sftp(bunyan.createLogger({ name: 'dummy' }), {
-    host: process.env.HOSTNAME,
-    port: process.env.PORT,
-    username: process.env.USER,
+    host: process.env.SFTP_HOSTNAME,
+     port: Number(process.env.PORT),
```
Comment (ElBabbler) [inline spec/triggers/read.spec.js]:
```suggestion
    port: Number(process.env.PORT),
```

**Pull request merged**